batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -0,0 +1,205 @@
+				            }
+			            }).withOutputParallelization(false)
+			            .withQuery(
+			                String.format("select uuid from %s where %s >= ? and %s < ?", this.tableName, tableId, tableId))
+			            .withRowMapper(new JdbcIO.RowMapper<String>() {
+				            
+				            @Override
+				            public String mapRow(ResultSet resultSet) throws Exception {
+					            return resultSet.getString("uuid");
+				            }
+			            }).withCoder(NullableCoder.of(StringUtf8Coder.of())));
+		}
+	}
+	
+	/**
+	 * batch together request using batchSize and generate segment descriptors `_id?<uuid>,<uuid>..`
+	 */
+	public static class CreateSearchSegments extends PTransform<PCollection<String>, PCollection<SearchSegmentDescriptor>> {
+		
+		private final String baseBundleUrl;
+		
+		private final String resourceType;
+		
+		private final int batchSize;
+		
+		public CreateSearchSegments(String resourceType, String baseBundleUrl, int batchSize) {
+			this.baseBundleUrl = baseBundleUrl;
+			this.batchSize = batchSize;
+			this.resourceType = resourceType;
+		}
+		
+		@Override
+		public PCollection<SearchSegmentDescriptor> expand(PCollection<String> uuIds) {
+			return uuIds
+			        // create KV required by GroupIntoBatches
+			        .apply(ParDo.of(new DoFn<String, KV<String, String>>() {
+				        
+				        @ProcessElement
+				        public void processElement(ProcessContext c) {
+					        if (c.element() != null) {
+						        c.output(KV.of(resourceType, c.element()));
+					        }
+				        }
+			        }))
+			        .apply(String.format("GroupIntoBatches of %s", this.batchSize),
+			            GroupIntoBatches.<String, String> ofSize(this.batchSize))
+			        .apply("Generate Segments", ParDo.of(new DoFn<KV<String, Iterable<String>>, SearchSegmentDescriptor>() {
+				        
+				        @ProcessElement
+				        public void processElement(@Element KV<String, Iterable<String>> element,
+				                OutputReceiver<SearchSegmentDescriptor> r) {
+					        List<String> uuIds = new ArrayList<String>();
+					        element.getValue().forEach(uuIds::add);
+					        r.output(SearchSegmentDescriptor.create(
+					            String.format("%s?_id=%s", baseBundleUrl, String.join(",", uuIds)), uuIds.size()));
+					        
+				        }
+			        }));
+		}
+	}
+	
+	public ResultSet FetchMaxId(String tableName) throws SQLException {
+		String tableId = tableName + "_id";
+		Connection con = this.comboPooledDataSource.getConnection();
+		Statement statement = con.createStatement();
+		ResultSet resultSet = statement.executeQuery(String.format("SELECT MAX(`%s`) from %s", tableId, tableName));
+		return resultSet;
+	}
+	
+	public JdbcIO.DataSourceConfiguration getJdbcConfig() {
+		return JdbcIO.DataSourceConfiguration.create(this.comboPooledDataSource);
+	}
+	
+	public Map<String, Integer> CreateIdRanges(int count, int rangeSize) {
```

Comment:
same here  createIdRanges(int count, int rangeSize)

Reply:
Thanks @gitcliff for the catch, yes it should be! Forgot to rename initially this was a class